tools/proofers/srchrep.js

```diff
@@ -8,7 +8,7 @@ var srchrep = (function() {
     }
 
     function escapeRegExp(string) {
-        return string.replace(/[.*+\-?^${}()|[\]\\]/g, '\\$&'); // $& means the whole matched string
+        return string.replace(/[.*+?^${}()|[\]\\]/g, '\\$&'); // $& means the whole matched string
```

Comment:
Testing shows that there's one other character in the replace string above that misbehaves when used as a s/r. More specifically, as a replacement string. The `$` doesn't behave as expected.

I've probably gone about my testing backwards, but first tests show that:
s: `$`, r: `$$` shows no visible effect.
However, s: `$`, r: `$$$$`, shows the replacement as `$$`. 

So, what's happening with the `$` is that it's only doing one of every two. I.e. in the first example, it probably was actually replacing the intital `$`, but since it was only taking one of the ones in the replacement string, it didn't appear that anything was happening.

Reply:
postscript: when I tried an odd number (5), the replacement string was 3 characters long.

Reply:
Interesting. I learn something new every day. Apparently the replace value has special "[replacement patterns](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/replace#Specifying_a_string_as_a_parameter)" all revolving around the '$' character. I've sent a fix, but this has been around forever it seems. I've sent a fix that avoids these replacement patterns by using a replacement function.